src/liquidationAuction.ml

```diff
@@ -578,57 +441,12 @@ let liquidation_auction_get_current_auction (auctions: liquidation_auctions) : c
  * returns the contents of the removed slice, the tree root the slice belonged to, and the updated auctions
 *)
 let pop_slice (auctions: liquidation_auctions) (leaf_ptr: leaf_ptr): liquidation_slice_contents * avl_ptr * liquidation_auctions =
-  let avl_storage = auctions.avl_storage in
-
-  (* pop the leaf from the storage *)
-  let leaf = avl_read_leaf avl_storage leaf_ptr in
-  let avl_storage, root_ptr = avl_del avl_storage leaf_ptr in
-
-  (* fixup burrow_slices *)
-  let burrow_slices = match Ligo.Big_map.find_opt leaf.contents.burrow auctions.burrow_slices with
-    | None -> (failwith "invariant violation: got a slice which is not present on burrow_slices": burrow_liquidation_slices)
-    | Some s -> s in
-  let burrow_slices =
-    match (leaf.older, leaf.younger) with
-    | (None, None) -> (* leaf *) (None: burrow_liquidation_slices option)
-    | (None, Some younger) -> (* leaf - younger - ... *) Some { burrow_slices with oldest_slice = younger }
-    | (Some older, None) -> (* .. - older - leaf *) Some { burrow_slices with youngest_slice = older }
-    | (Some _, Some _) -> (* ... - leaf - ... *) Some burrow_slices
-  in
-
-  (* fixup older and younger pointers *)
-  let avl_storage = (
-    match leaf.younger with
-    | None -> avl_storage
-    | Some younger_ptr ->
-      avl_update_leaf
-        avl_storage
-        younger_ptr
-        (fun (younger: liquidation_slice) ->
-           assert (younger.older = Some leaf_ptr);
-           { younger with older = leaf.older }
-        )
-  ) in
-  let avl_storage = (
-    match leaf.older with
-    | None -> avl_storage
-    | Some older_ptr ->
-      avl_update_leaf
-        avl_storage
-        older_ptr
-        (fun (older: liquidation_slice) ->
-           assert (older.younger = Some leaf_ptr);
-           { older with younger = leaf.younger }
-        )
-  ) in
-
-  (* return *)
-  ( leaf.contents
+  let element, burrow_slices = slice_list_from_leaf_ptr auctions leaf_ptr in
+  let avl_storage, burrow_slices, root_ptr, contents = slice_list_remove burrow_slices auctions.avl_storage element in
+  let auctions = {auctions with avl_storage=avl_storage;} in
```

Comment:
I saw this pattern (we pass `avl_storage` and `slice_list` around, then put the `avl_storage` back into the auctions and call `slices_to_auction_state`) multiple times, maybe this means that `avl_storage` needs to be in `SliceList`?